Userland/Libraries/LibHTTP/Job.cpp

```diff
@@ -244,12 +329,16 @@ void Job::on_socket_connected()
                     m_content_length = length.value();
             }
             dbgln_if(JOB_DEBUG, "Job: [{}] = '{}'", name, value);
-            return;
+            if (!MUST(m_socket->can_read_without_blocking())) {
+                dbgln_if(JOB_DEBUG, "Can't read headers anymore, byebye :(");
+                return;
+            }
```

Comment:
Is it guaranteed that we'll receive all headers before blocking?

Reply:
no, but this will be invoked later when we have more data and we'll resume reading headers.